src/components/Login.tsx

```diff
@@ -16,24 +17,48 @@ const Login = ({ loginWalletOptions }: LoginProps): JSX.Element => {
   const [loading, startLoading] = React.useState<boolean>(false);
   const [alertError, setAlertError] = React.useState<string>("");
   const [popoverVisible, setPopoverVisible] = React.useState<boolean>(false);
+  const [registrationVisible, setRegistrationVisible] = React.useState<boolean>(
+    false
+  );
+
+  const [walletAddress, setWalletAddress] = React.useState<string>("");
+  const [walletType, setWalletType] = React.useState<wallet.WalletType>(
+    loginWalletOptions || wallet.WalletType.NONE
+  );
 
   const dispatch = useAppDispatch();
   const userId = useAppSelector((state) => state.user.id);
-  const walletType = useAppSelector((state) => state.user.walletType);
+
+  const setLoginAndSession = (fromId: string) => {
+    setRegistrationVisible(false);
+    dispatch(userLogin({ id: fromId, walletType }));
+    session.saveSession({ id: fromId, walletType });
+  };
+
+  const resetLoginAndSession = (e: Error) => {
+    setRegistrationVisible(false);
+    setAlertError(e.message);
+    setWalletAddress("");
+    setWalletType(wallet.WalletType.NONE);
+    session.saveSession({ id: undefined, walletType });
+  };
 
   const login = async (walletType: wallet.WalletType) => {
     if (loading) return;
     startLoading(true);
+    setWalletType(walletType);
     try {
-      const walletAddress = await wallet.wallet(walletType).login();
+      const walletAddrLocal = await wallet.wallet(walletType).login();
       sdk.setupProvider(walletType);
-      const fromId = await sdk.getSocialIdentity(walletAddress);
+      const fromId = await sdk.getSocialIdentity(walletAddrLocal);
       if (fromId) {
-        dispatch(userLogin({ id: fromId, walletType }));
-        session.saveSession({ id: fromId, walletType });
+        setLoginAndSession(fromId);
+      } else {
+        setWalletAddress(walletAddrLocal);
+        setRegistrationVisible(true);
       }
```

Comment:
this can be removed from setLoginAndSession and added after the else setRegistrationVisible(true);